inception-recommendation-api/src/test/java/de/tudarmstadt/ukp/inception/recommender/api/evaluation/EvaluationResultTest.java

```diff
@@ -0,0 +1,135 @@
+        EvaluationResult calc = new EvaluationResult(null, instances.stream());
+        assertThat(calc.computeAccuracyScore()).as("accuracy is correctly calculated")
+                .isEqualTo(4.0 / 9.0);
+    }
+
+    @Test
+    public void thatPrecisionWorks()
+    {
+        EvaluationResult calc = new EvaluationResult(null, instances.stream());
+        assertThat(calc.computePrecisionScore()).as("precision is correctly calculated")
+                .isEqualTo((0.5 + 0.5 + 1.0 / 3.0) / 3);
+    }
+
+    @Test
+    public void thatRecallWorks()
+    {
+        EvaluationResult calc = new EvaluationResult(null, instances.stream());
+        assertThat(calc.computeRecallScore()).as("recall is correctly calculated")
+                .isEqualTo((0.5 + 0.5 + 1.0 / 3.0) / 3);
+    }
+
+    @Test
+    public void thatF1Works()
+    {
+        EvaluationResult calc = new EvaluationResult(null, instances.stream());
+        assertThat(calc.computeF1Score()).as("f1 is correctly calculated")
+                .isEqualTo(2 * (4.0 / 9.0 * 4.0 / 9.0) / (8.0 / 9.0));
+    }
+    
+    @Test
+    public void thatIgnoringALabelWorks()
+    {
+        EvaluationResult calc = new EvaluationResult(asList("PER"), instances.stream());
+        assertThat(calc.computeF1Score()).as("f1 with ignore label is correctly calculated")
+                .isEqualTo(2 * (0.5 * (0.5 + 1.0 / 3) * 0.5)
+                        / (0.5 + (0.5 + 1.0 / 3) * 0.5));
+        assertThat(calc.computeRecallScore()).as("recall with ignore label is correctly calculated")
+                .isEqualTo(0.5);
+        assertThat(calc.computeAccuracyScore())
+                .as("accuracy with ignore label is correctly calculated").isEqualTo(6.0 / 12);
+        assertThat(calc.computePrecisionScore())
+                .as("precision with ignore label is correctly calculated")
+                .isEqualTo((0.5 + 1.0 / 3) * 0.5);
+    }
+
+    @Test
+    public void thatNumOfLabelsWorks()
+    {
+        EvaluationResult calc = new EvaluationResult(asList(), instances.stream());
+        assertThat(calc.getNumOfLabels()).as("check num of labels for no ignoreLabel").isEqualTo(3);
+
+        calc = new EvaluationResult(asList("PER"), instances.stream());
+        assertThat(calc.getNumOfLabels()).as("check num of labels for one ignoreLabel")
+                .isEqualTo(2);
+
+        calc = new EvaluationResult(asList("PER", "ORG"), instances.stream());
+        assertThat(calc.getNumOfLabels()).as("check num of labels for two ignoreLabel")
+                .isEqualTo(1);
+
+    }
+
+    @Test
+    public void thatMissingClassesWorks()
+    {
+        // test with classes which are never gold or never predicted
+        List<AnnotatedTokenPair> newInstances = new ArrayList<>(instances);
+        newInstances.add(new AnnotatedTokenPair("PART", "ORG"));
+        newInstances.add(new AnnotatedTokenPair("PER", "PUNC"));
+        EvaluationResult calc = new EvaluationResult(null, newInstances.stream());
```

Comment:
I really do not like passing null into constructors, you should create a dedicated constructor for it and avoid null as much as possible.